### PR TITLE
Add types from unofficial-builds

### PR DESCRIPTION
**Pull request by rvagg**
Ref: https://github.com/nodejs/unofficial-builds/issues/23

I don't think this is the _proper_ fix, but it's a good enough one for now. The problem will be that adding new unofficial-build types is going to require messing in here and that's going to be forgotten until someone opens an issue to point out the discrepancy. I'm not sure what the best option is but maybe there's a way to do a _"best guess where we don't recognise the type"_ that's maybe a flag that we run for unofficial-builds but not for official releases (we'd want to be a lot more intentional there). For now this will do.

### CODE REVIEW

**Review by richardlau — APPROVED** [review]

**Comment by richardlau** [comment]
> 
> 
> Ref: [nodejs/unofficial-builds#23](https://github.com/nodejs/unofficial-builds/issues/23)
> 
> I don't think this is the _proper_ fix, but it's a good enough one for now. The problem will be that adding new unofficial-build types is going to require messing in here and that's going to be forgotten until someone opens an issue to point out the discrepancy. I'm not sure what the best option is but maybe there's a way to do a _"best guess where we don't recognise the type"_ that's maybe a flag that we run for unofficial-builds but not for official releases (we'd want to be a lot more intentional there). For now this will do.

I was thinking maybe allow the types map to be specified/overridden in a json file and then we could put the one for the unofficial builds in the unofficial-builds repo so that it would at least be alongside the scripts that create the artifacts.

File: transform-filename.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ const types = {
   'linux-armv6l': 'linux-armv6l',
   'linux-x64': 'linux-x64',
   'linux-x86': 'linux-x86',
+  'linux-x64-musl': 'linux-x64-musl',
```
Comment (nschonni) [inline transform-filename.js:13]: Does it make sense to call it out here?
```suggestion
  'linux-x64-musl': 'linux-x64-musl', // unofficial builds
```

**Review by nschonni — COMMENTED** [review]
The change makes sense to me, but when I was looking at it yesterday, I was wondering if the `types` should get sorted

**Comment by rvagg** [comment]
OK, sorted the list but put an `// unofficial-builds:` section at the bottom. I've also added the win-arm64 types as they'll hopefully become more regular @joaocgreis got one up for the latest https://unofficial-builds.nodejs.org/download/release/v14.13.0/

It's unfortunate that we've included the extension on the windows file types as they're omitted from the others. I'd rather they not be but we're a bit stuck with that pattern now, so be it.

**Comment by rvagg** [comment]
ok, published as 1.4.0, installed on the main web server and it made no difference to index.tab and index.json there, installed on unofficial-builds and we now have file types:

```json
{"version":"v14.13.1","date":"2020-10-07","files":["headers","linux-armv6l","linux-x64-musl","linux-x64-pointer-compression","linux-x64-usdt","win-arm64-7z","win-arm64-zip"],"npm":"6.14.8","v8":"8.4.371.19","uv":"1.40.0","zlib":"1.2.11","openssl":"1.1.1g","modules":"83","lts":false,"security":false},
```

https://unofficial-builds.nodejs.org/download/release/index.json
https://unofficial-builds.nodejs.org/download/release/index.tab

**Pull request merged**